### PR TITLE
LPS-128998 [Content Performance] Separate dispatch from state and unified warnings

### PR DESCRIPTION
**Pull request by dgarciasarai**
**Description**

This is the third pull request to fix [LPS-128998 Content Performance available endpoints should load at the same time](https://issues.liferay.com/browse/LPS-128998) as I said in a previous PR (see `Notes` from description, but in summary, the idea is that every component reads data from the store instead of calling its `dataProvider` and wait until all endpoints are resolved to display the data in the UI): https://github.com/liferay-frontend/liferay-portal/pull/885#issue-588734880.

As the pull can be huge, I'm going to create sub-tasks for the issue and send the code for each of them separately.

**How to test it**

- Connect Liferay DXP with Analytics Cloud
- Create a Content Page
- Wait 24 hours to let Analytics Cloud process the data (if you don't want to wait for it, change the date of your computer and create the page in the past 😉  )
- Open the Content Performance panel

**Within this pull request**

Everything should work as before. No UI changes.
- Separate dispatch from state in StoreContext
- Unified historical warning and warning (we are not displaying different messages, so a simple warning is enough)
- Use dispatch to call `addWarning` action

**Final solution for the issue**

In the future the handlers from `Navigation` component will be converted into actions and dispatch it when necessary. Also, some information will be removed from the prop-drilling and save it in the store as-well. Components will read from the store to display the data.

**History pull requests**

- LPS-128998 [Content Performance] Refactor APIService to be called in any component: https://github.com/brianchandotcom/liferay-portal/pull/99816
- LPS-128998 [Content Performance] Save attributes in store instead of prop-drilling them: https://github.com/brianchandotcom/liferay-portal/pull/99883

### CODE REVIEW

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/StoreContext.js [hunk 1]
```diff
@@ -9,67 +9,47 @@
...
-	}, [dispatch]);
-
-	const hasWarning = state.warning;
-
-	return [hasWarning, addWarning];
-}
+/**
+ * @see https://react-redux.js.org/api/hooks#usedispatch
+ */
+export const useDispatch = () => useContext(StoreDispatchContext);
```
Comment (wincent) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/StoreContext.js:55]: I understand you're following the Redux pattern here, and we also have a `useDispatch` hook in `layout-content-page-editor-web`, but:

1. You're not actually using Redux.
2. This hook literally does nothing other than call another hook.

Which means that you could just call the other hook, and avoid an unnecessary layer of indirection. FWIW, see [this commit in `layout-content-page-editor-web` to compare what it used to look like before](https://github.com/liferay/liferay-portal/commit/2f2d2f67ccbb9803b2b5cd36760f138bcb253b8d).

But this is non-blocking feedback...

just-sayin

In general I'd be wary of bringing in too much Redux-like boilerplate before you really need it (like the separate files for the actions and the action types, when you currently only have one action, at least as of this PR). One of the main complaints people have about Redux is [the boilerplate](https://redux.js.org/recipes/reducing-boilerplate).

**Review by wincent — APPROVED** [review]
Feel free to forward once we see CI green.

**Comment by dgarciasarai** [comment]
Hey @wincent, I'll simplify the dispatch, I remove the hook `useDispatch` and use it directly in the components as:

```js
const dispatch = useContext(StoreDispatchContext);
```

Let me know if you see something else. If not, forward it directly :) Thanks!

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by wincent** [comment]
Going to forward, but due to the force-push the bot will surely re-run the tests.

whatever

**Comment by wincent** [comment]
ci:forward

**Comment by dgarciasarai** [comment]
> Going to forward, but due to the force-push the bot will surely re-run the tests.

Yes, no problem! And thanks again, frontend friend :)

**Pull request closed without merging**